cmd/kubelet/app/server.go

```diff
@@ -412,13 +429,222 @@ func run(s *options.KubeletServer, kcfg *KubeletConfig) (err error) {
 	return nil
 }
 
+// bootstrapClientCert will request a client cert for kubelet if '--bootstrapKubeconfig'
+// is non-empty and no client certificate is available in kubeconfig.
+// On success, the result certificate and key file will be stored on tmpfs (default to /var/run/kubernetes/),
+// and the kubeconfig will point to those files.
+// If '--bootstrapKubeconfig' is set but kubeconfig file doesn't exist, then the function will fail.
```

Comment:
If kubeconfig doesn't exist, we need to know/define `cluster name`, `user name`, I don't know what's the best default values for them, so I leave a TODO here.